tutorial6_hierarchical/demo/main.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * CONSTANTS AND GLOBALS
+ * */
+const width = window.innerWidth * 0.9,
+  height = window.innerHeight * 0.7,
+  margin = { top: 20, bottom: 50, left: 60, right: 40 },
+  tooltipSpecs = { width: 100, height: 50 };
+
+let svg;
+let tooltip;
+
+/**
+ * APPLICATION STATE
+ * */
+let state = {
+  data: null,
+  hover: null,
+};
+
+/**
+ * LOAD DATA
+ * */
+d3.json("../../data/flare.json", d3.autotype).then(data => {
+  state.data = data;
+  init();
+});
+
+/**
+ * INITIALIZING FUNCTION
+ * this will be run *one time* when the data finishes loading in
+ * */
+function init() {
+  const container = d3.select("#d3-container").style("position", "relative");
+
+  tooltip = container
+    .append("div")
+    .attr("width", 100)
+    .attr("height", 100)
+    .style("position", "absolute")
+    .style("background-color", "white");
+
+  svg = container
+    .append("svg")
+    .attr("width", width)
+    .attr("height", height);
+
+  // groups the data by genre, type and rating
+  // make hierarchy
+  const root = d3
+    .hierarchy(state.data) // children accessor
+    .sum(d => d.value) // sets the 'value' of each level
```

Comment:
updated this - before it wan't aggregating properly, just wanted to bring your attention to it